addons/eventbus/src/main/java/org/vaadin/spring/events/internal/MethodListenerWrapper.java

```diff
@@ -98,11 +98,14 @@ public boolean supports(Event<?> event) {
             if (listenerMethod.isAnnotationPresent(EventBusListenerMethod.class)) {
                 EventBusListenerMethod annotation = listenerMethod.getAnnotation(EventBusListenerMethod.class);
                 EventBusListenerMethodFilter filter = annotation.filter().newInstance();
+                String target = annotation.target();
                 EventScope scope = annotation.scope();
                 if (scope.equals(EventScope.UNDEFINED)) {
                     scope = event.getScope();
                 }
-                supports = supports && filter.filter(event.getPayload()) && event.getScope().equals(scope);
+                supports = supports && filter.filter(event.getPayload())
+                        && event.getScope().equals(scope)
+                        && event.getTarget().startsWith(target);
```

Comment:
I don't think this should be the default behaviour. I'd expect the method to by default match the entire target. We could add an additional flag to the annotation that instructs the method listener wrapper to only match the beginning of the string.